debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -103,18 +105,18 @@ public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPoi
 		boolean lockwordPrinted = false;
 		UDATA nestedFieldOffset = new UDATA(0);
 		boolean flatObject = false;
-
 		J9ClassPointer nestedClassHierarchy[] = null;
+		boolean showObjectHeader = tabLevel <= 1; /* This is true when !flatobject is specified are currently printing a nested structure */
 
 		if ((null != nestingHierarchy) && (nestingHierarchy.length > 0)) {
 			flatObject = true;
 		}
 
-		/* This is true when !flatobject is specified are currently printing a nested structure */
-		boolean showObjectHeader = tabLevel <= 1;
-
+		/**
+		 * Given we have a flatObject, conditionally increment ahead to the correct object index if we have a flattened array, 
+		 * and skip ahead to the specified field by its offset
+		 */
```

Comment:
Please reserve javadoc comments for features (types, members) that need documentation; just use a simple multi-line comment here.

Also be consistent with punctuation: if the comment is a sentence (it starts with a capital), it should end with a period.